compiler/injectdestructors.nim

```diff
@@ -1205,7 +1224,11 @@ proc moveOrCopy(dest, ri: PNode; c: var Con; s: var Scope, flags: set[MoveOrCopy
     of nkRaiseStmt:
       result = pRaiseStmt(ri, c, s)
     else:
-      if isAnalysableFieldAccess(ri, c.owner) and isLastRead(ri, c, s) and
+      let isOwnsData = ownsData(c, s, ri2, flags)
```

Comment:
Arbitrary logic. Shouldn't `isAnalysableFieldAccess` etc be able to handle this? I think there is a real bug to be fixed here somewhere, not more custom logic to be added.